workflow/controller/operator.go

```diff
@@ -1231,17 +1250,33 @@ func getTemplateOutputsFromScope(tmpl *wfv1.Template, scope *wfScope) (*wfv1.Out
 }
 
 func (woc *wfOperationCtx) executeScript(nodeName string, tmpl *wfv1.Template, boundaryID string) *wfv1.NodeStatus {
+	var skipNodeInitialization bool
 	node := woc.getNodeByName(nodeName)
 	if node != nil {
-		return node
+		if strings.Contains(node.Message, exceededQuota) {
+			skipNodeInitialization = true
+			woc.log.Infof("Node %s exists but a pod does not: %s", nodeName, node.Message)
+		} else {
+			return node
+		}
 	}
 	mainCtr := tmpl.Script.Container
 	mainCtr.Args = append(mainCtr.Args, common.ExecutorScriptSourcePath)
+	var messages []string
 	_, err := woc.createWorkflowPod(nodeName, mainCtr, tmpl)
 	if err != nil {
-		return woc.initializeNode(nodeName, wfv1.NodeTypePod, tmpl.Name, boundaryID, wfv1.NodeError, err.Error())
+		if strings.Contains(err.Error(), exceededQuota) {
```

Comment:
Move the strings.Contains call into a function called ExceededQuota or something?